File: android/app/src/main/java/com/opentokreactnative/OTRN.java
package com.opentokreactnative;

import com.opentok.android.Session;
import com.opentok.android.Stream;
import java.util.concurrent.ConcurrentHashMap;

public class OTRN {

    public static OTRN sharedState;

    private ConcurrentHashMap<String, Stream> subscriberStreams = new ConcurrentHashMap<>();
    private ConcurrentHashMap<String, Session> sessions = new ConcurrentHashMap<>();

    public static synchronized OTRN getSharedState() {

        if (sharedState == null) {
            sharedState = new OTRN();
        }
        return sharedState;
    }

    public ConcurrentHashMap<String, Stream> getSubscriberStreams() {

        return this.subscriberStreams;
    }


    public ConcurrentHashMap<String, Session> getSessions() {

        return this.sessions;
    }

    private OTRN() {}
}

File: android/app/src/main/java/com/opentokreactnative/OTRN.kt
package com.opentokreactnative

import com.opentok.android.Session
import com.opentok.android.Stream

public final class OTRN {

  companion object {
    lateinit var session: Session
    lateinit var stream: Stream

    fun putSession(s : Session) {
        this.session = s
    }

    fun getSess() : Session {
        return this.session
    }

    fun putStream(s : Stream) {
        this.stream = s
    }

    fun getStr() : Stream {
        return this.stream
    }
  }
}